### ISSUE TITLE
Allow namespaced migration files

### ISSUE THREAD

**Issue by dereuromark** [cakephp/migrations#475]
I wonder if we could use https://github.com/cakephp/phinx/issues/1820 to namespace our migration classes
They are the only code I still have to exclude from sniffers so far:

```
FOUND 2 ERRORS AFFECTING 1 LINE
----------------------------------------------------------------------------------------------------------------------------------------
 6 | ERROR | Class name "MigrationLanguages" doesn't match filename, expected "20201014000218_MigrationLanguages"
 6 | ERROR | Each class must be in a namespace of at least one level (a top-level vendor name)
```

With the namespace way it would also probably be a bit cleaner and less conflicting regarding the names.

Right now I always prefix them with Migration to avoid naming clashes/collisions with existing top level classes.

**Comment by dereuromark** [comment]
Done with https://github.com/cakephp/cakephp-codesniffer/pull/320

**Comment by dereuromark** [comment]
laravel also spaces the timestamp datetime for readability:

    2024_07_15_203230_CreateProductsTable

We could do sth similar when we refactor this into class namespaced or alike.

**Comment by markstory** [comment]
Revisiting the filenames and namespacing of migrations are both things I'd like to follow up on once phinx is removed. Doing it before that time will create compatibility issues.

**Comment by LordSimal** [comment]
I'd rather use anonymous classes and no namespaces (like e.g. Laravel does).
We can for sure make CS work with that as well.

**Comment by dereuromark** [comment]
I agree that if we move to anonm. class and no namespace, we could just as well move them into

    resources/

folder instead of config/

We have to find a way to make it CS/PHPStan compatible, though without having to silence a lot.

**Comment by umer936** [comment]
> 2024_07_15_203230_

I like this.

**Comment by dereuromark** [comment]
https://github.com/cakephp/migrations/pull/882

**Comment by dereuromark** [comment]
PR open.
